bin/runtime-common/src/messages_call_ext.rs

```diff
@@ -291,7 +302,7 @@ impl<
 				return sp_runtime::transaction_validity::InvalidTransaction::Stale.into()
 			},
 			Some(CallInfo::ReceiveMessagesDeliveryProof(proof_info))
-				if proof_info.is_obsolete() =>
+				if is_pallet_halted || proof_info.is_obsolete() =>
```

Comment:
Is this needed ? If `is_pallet_halted == true`  and `self.call_info() == Some(_)` won't the first branch of this match be executed (`Some(proof_info) if is_pallet_halted => {...}`) ?